backend/endpoints/rooms.py

```diff
@@ -353,3 +353,92 @@ async def close_conversation(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="unable to close conversation",
     )
+
+
+@router.get(
+    "/org/{org_id}/rooms/{room_id}/members",
+    response_model=ResponseModel,
+    status_code=status.HTTP_200_OK,
+    responses={
+        400: {"detail": "Invalid keyword supplied"},
+        404: {"detail": "Room not found"},
+        424: {"detail": "Failure to retrieve room members"},
+    },
+)
+async def get_members(org_id: str, room_id: str, keyword: Optional[str] = None):
+
+    """Get room members.
+    Returns all members in a room if the room is found in the database
+    Returns filtered members in a room if keyword is supplied
+    Raises HTTP_400_BAD_REQUEST if keyword supplied is not admin or member
+    Raises HTTP_404_NOT_FOUND if the room is not found
+    Raises HTTP_424_FAILED_DEPENDENCY if there is an error retrieving the room members
+    Args:
+        org_id (str): A unique identifier of an organisation
+        room_id (str): A unique identifier of the room
+    Returns:
+        HTTP_200_OK (Room members retrieved successfully):
+
+        {
+            "status": "success",
+            "message": "Room members retrieved",
+            "data": {
+                "61696f5ac4133ddaa309dcfe": {
+                "closed": false,
+                "role": "admin",
+                "starred": false
+                },
+                "6169704bc4133ddaa309dd07": {
+                "closed": false,
+                "role": "admin",
+                "starred": false
+                }
+            }
+        }
+
+    Raises:
+        HTTPException [400]: Invalid keyword supplied
+        HTTPException [404]: Room not found
+        HTTPException [424]: Failure to retrieve room members
+    """
+    room = await get_room(org_id, room_id)
+
+    if not room:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Room doesn't exist"
+        )
+
+    members = room["room_members"]
+    if members and members.get("status_code") is not None:
+        raise HTTPException(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY,
+            detail="Failure to retrieve room members",
+        )
+
+    if keyword:  # if keyword is supplied
```

Comment:
Well done boss @pauline-banye 

So the filter option is to only return members or admins in a room, i thought @o4codes was referring to filtering based starred, close and roles, maybe u can include these as well boss.

Reply:
Repent o @man-of-mind re filtering for starred and closed rooms, I had a conversation with both @o4codes and @zxenonx and it was concluded that there's no use case for it. 

Its should be done in the room if at all. They are both Boolean fields without any specific information. That's why I ended up leaving them out.

Reply:
Oh...okay boss

Reply:
Do you have a different opinion? Is there a use case you can think of?

Reply:
Not really o.....my ogas don talk

Reply:
If you do have an opinion or a change you want me to implement, pls speak up.

Reply:
😂😂😂😂

No boss, I am serious

Reply:
I feel when an endpoint is created, the possibilities of thaet endpoint should be maxed out.
The concepts of use cases helps in aligning priority.
I feel those extra features could still be added.
But then for us to move faster, it can be waved off, till someone free enough decides to add them

Reply:
I don't think we need to filter based on these parameters (closed, starred) for now.